oap-server/analyzer/meter-analyzer/src/main/java/org/apache/skywalking/oap/meter/analyzer/prometheus/rule/Rules.java

```diff
@@ -40,36 +49,76 @@
         return loadRules(path, Collections.emptyList());
     }
 
-    public static List<Rule> loadRules(final String path, List<String> enabledRules) throws ModuleStartException {
-        File[] rules;
-        try {
-            rules = ResourceUtils.getPathFiles(path);
-        } catch (FileNotFoundException e) {
-            throw new ModuleStartException("Load fetcher rules failed", e);
-        }
-        return Arrays.stream(rules)
-            .filter(File::isFile)
-            .map(f -> {
-                try (Reader r = new FileReader(f)) {
-                    String fileName = f.getName();
-                    int dotIndex = fileName.lastIndexOf('.');
-                    fileName = (dotIndex == -1) ? fileName : fileName.substring(0, dotIndex);
-                    if (!enabledRules.contains(fileName)) {
-                        return null;
+    public static List<Rule> loadRules(final String path, List<String> enabledRules) throws IOException {
+
+        final Path root = ResourceUtils.getPath(path);
+
+        Map<String, Boolean> formedEnabledRules = enabledRules
+                .stream()
+                .map(rule -> {
+                    rule = rule.trim();
+                    if (rule.startsWith("/")) {
+                        rule = rule.substring(1);
+                    }
+                    if (!rule.endsWith(".yaml") && !rule.endsWith(".yml")) {
+                        return rule + "{.yaml,.yml}";
                     }
-                    Rule rule = new Yaml().loadAs(r, Rule.class);
-                    rule.setName(fileName);
                     return rule;
-                } catch (IOException e) {
-                    LOG.debug("Reading file {} failed", f, e);
-                }
+                })
+                .collect(Collectors.toMap(rule -> rule, $ -> false));
+        List<Rule> rules;
+        try (Stream<Path> stream = Files.walk(root)) {
+            rules = stream
+                    .filter(it -> formedEnabledRules.keySet().stream()
+                                    .anyMatch(rule -> {
+                                        boolean matches = FileSystems.getDefault().getPathMatcher("glob:" + rule)
+                                                .matches(root.relativize(it));
+                                        if (matches) {
+                                            formedEnabledRules.put(rule, true);
+                                        }
+                                        return matches;
+                                    }))
+                    .map(pathPointer -> {
+                        Path relativize = root.relativize(pathPointer);
+                        String relativizePath = relativize.toString();
+                        relativizePath = relativizePath.substring(0, relativizePath.lastIndexOf("."));
+                        return getRulesFromFile(relativizePath, pathPointer);
```

Comment:
```suggestion
                        // Use relativized file path without suffix as the rule name. 
                        String ruleName = root.relativize(pathPointer).toString().substring(0, relativizePath.lastIndexOf("."));
                        return getRulesFromFile(ruleName, pathPointer);
```

I think this is better?